libraries/pybamm-developer/03_spm.md

```diff
@@ -0,0 +1,373 @@
+and
+[`pybamm.arcsinh`](https://docs.pybamm.org/en/stable/source/api/expression_tree/functions.html#pybamm.arcsinh)
+functions:
+
+```python
+four = pybamm.Scalar(4)
+two = pybamm.sqrt(four)
+arcsinh_four = pybamm.arcsinh(four)
+```
+
+You can see a list of all the functions available in PyBaMM in the [documentation](https://docs.pybamm.org/en/stable/source/api/expression_tree/functions.html).
+
+::::challenge{id=spm-output-variables, title=SPM output variables}
+
+Define the output variables for the SPM model. You will need to define the
+overpotentials $\eta_n$ and $\eta_p$, the exchange current densities $i_{0,n}$
+and $i_{0,p}$, and the terminal voltage $V$. You will also need to define the
+OCPs $U_n$ and $U_p$ as functions of the surface stoichiometries $x_n^s$ and
+$x_p^s$. You can use `pybamm.FunctionParameter` to define the OCPs as functions
+of the surface stoichiometries.
+
+Define the following output variables for the model
+- Terminal voltage $V$
+- Surface concentration in negative particle $c_n^s$
+
+:::solution
+```python
+# call universal constants (PyBaMM has them built in)
+R = pybamm.constants.R
+
+# define temperature
+T = pybamm.Parameter("Temperature [K]")
+
+# define new parameters for the output variables
+c_e = pybamm.Parameter("Electrolyte concentration [mol.m-3]")
+k_i = [pybamm.Parameter(f"Reaction rate constant for {d} [m.s-1]") for d in domains]
+c_i_max = [pybamm.Parameter(f"Maximum concentration for {d} [mol.m-3]") for d in domains]
+
+# define intermediate variables and OCP function parameters
+c_i_s = [pybamm.surf(c_i[i]) for i in [0, 1]]
+x_i_s = [c_i_s[i] / c_i_max[i] for i in [0, 1]]
+i_0_i = [k_i[i] * F * (pybamm.sqrt(c_e) * pybamm.sqrt(c_i_s[i]) * pybamm.sqrt(c_i_max[i] - c_i_s[i])) for i in [0, 1]]
+eta_i = [2 * R * T / F * pybamm.arcsinh(j_i[i] * F / (2 * i_0_i[i])) for i in [0, 1]]
+U_i = [pybamm.FunctionParameter(f"Open circuit potential for {d}", {"stoichiometry": x_i_s[i]}) for (i, d) in enumerate(domains)]
+
+# define output variables
+[U_n_plus_eta, U_p_plus_eta] = [U_i[i] + eta_i[i] for i in [0, 1]]
+V = U_p_plus_eta - U_n_plus_eta
+model.variables = {
+  "Terminal voltage [V]": V,
+  "Surface concentration in negative particle [mol.m-3]": c_i_s[0],
+}
+```
+:::
+::::
+
+## Discretising and solving the Single Particle Model
+
+Now that we have defined the SPM model, we can discretise and solve it using
+PyBaMM. We can use the same meshing and discretisation methods as in the
+previous section, but we now need to discretise the model on two different
+spatial domains.
+
+::::challenge{id=spm-discretise-solve, title=Discretise and solve the SPM model}
+
+Discretise and solve the SPM model using the same methods as in the previous
+section. The following parameter values object copies the parameters from the PyBaMM
+Chen2020 model, feel free to use this to define the parameters for the SPM model.
+    
+```python
+p = pybamm.ParameterValues("Chen2020")
+param = pybamm.ParameterValues(
```

Comment:
What's the motivation for doing this rather than just using the names we already use? This seems clunky IMO.